### DISCUSSION TITLE
Awesome Plugins Ideas

### DISCUSSION THREAD

**Discussion by Anu-bhav** [Flow-Launcher/Flow.Launcher#909] [Ideas]
Hi everyone,

I wanted to know if there is a way to get ulauncher plugins to work on flow launcher. I think most of them, if not all, are developped in python.

Not all of them, of course, as there are some exclusive for certain linux distros but there are some really nice to have plugins that enhances the launcher's experience as a whole. 

Some of them are: 
1. vs workspaces - https://github.com/barathbheeman/vsw-extension
2. Calculate Anything - https://github.com/tchar/ulauncher-albert-calculate-anything
3. UNotion - https://github.com/sergius02/ulauncher-unotion
4. Speedtest - https://github.com/KarlFrederik/ulauncher-speedtest
5. NordVPN - https://github.com/plibither8/ulauncher-nordvpn
6. Coingecko Query - https://github.com/NCRoxas/coingecko-query
7. Obsidian - https://github.com/mikebarkmin/ulauncher-obsidian
8. Youtube Search - https://github.com/NastuzziSamy/ulauncher-youtube-search
9. Shortcuts - https://github.com/brpaz/ulauncher-shortcuts
10.  DevDocs - https://github.com/brpaz/ulauncher-devdocs
11. OpenSubtitles - https://github.com/diogox/ulauncher-opensubtitles
12. HackerNews - https://github.com/diogox/ulauncher-hackernews
13. jump - https://github.com/USER0308/ulauncher-plugin-jump

I tried to omit all the ones which were already implemented or were too dependent on linux.

As you can see most of them either scrape the information needed from website, or they are dependent on a certain software to be installed.

It would be great if someone can find a solution to how it can be done without rewritting the plugin to work on flow launcher.

**Comment by onesounds** [comment]
Team members checked the plugin's structure, but it's a different from Flow.
It seems difficult to port easily. I'm sorry that I can't answer positively.  and If you let me know the plug-in you use often, other team members will consider developing it first. We will refer to the list you told us.

p.s Some of the plugins you mentioned are under development or are in plan. 
- under development: obsidian, vs code (from powertoy)
- plan : youtube search (Since it is already possible in the web search plug-in, we are considering the degree to which thumbnails are added.)
 Among the things we don't have in our plan, vpn and speed tests seem useful.

**Reply by Anu-bhav** [reply]
1) if possible, kindly make a plugin to support basic soundcloud(parse urls{playlist, single track}, play, pause, next, previous)
A useful library for parsing soundcloud urls: https://github.com/3jackdaws/soundcloud-lib

2) Instead of devdocs, kindly support https://github.com/chubin/cheat.sh. 
I often use it to know how to use commands or to find how to use a syntax.

3) I have an idea of a type of plugin that can be configured to use in different context.
 - we can map a software to open a specific(default) folder(a bit like vs workspace)
 - however we can specify which software we want to use with which folder.
   - VLC/mpv for music/video folder
   - typora/obsidian for folder containing markdown files
   - sublime text/notepad++/vscode for files in a folder we often modify
 - It can be launched using the prefix `d` for default and the app name, it will then open the app in that location only.
   - d vlc  -> autocompletes if we have multiple options for the same app, lets say 1st option is music folder, 2nd is video folder
   - d typora -> automatically opens typora in the specified folder.

4) VPN - for now I'm using nordvpn and I've mapped win+n to connect and disconnect to nord.
I'm currently using autohotkey to achieve this.

**Reply by deefrawley** [reply]
For Nord, if you have a command line that can be used (i.e. the one with AHK) then map this under the Plugin Runner plugin. This would need to be a fixed command line though so if you are doing something to dynamically choose the country to connect to then it won't work unless you setup multiple Plugin Runner shortcuts.

**Reply by Anu-bhav** [reply]
Hi @deefrawley,

It is a fixed command line, however I did set it up in a way that it can toggle on and off with the same key bind and it displays a notification status as well. 

It's nothing fancy, but it gets the job done.

![image](https://user-images.githubusercontent.com/41940586/147235728-aec75216-f536-485e-8c4f-2e95319cecef.png)
![image](https://user-images.githubusercontent.com/41940586/147235759-b8e5a168-d259-4a7a-bfe6-5d1f948b2ed2.png)
![image](https://user-images.githubusercontent.com/41940586/147235769-4ac4d7a3-62e3-4fea-98b4-1259ce57990c.png)


I do not think I will try it with nordvpn, however I try to test the Plugin Runner with openvpn as I do connect to other VPNs.
![image](https://user-images.githubusercontent.com/41940586/147238108-f05c9cff-b416-4d12-980e-39dfe7074a84.png)

**Comment by onesounds** [comment]
obsidian plugin is release. check the plugin store please.
